src/Cake.Common/Tools/VSWhere/VSWhereAliases.cs

```diff
@@ -0,0 +1,246 @@
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPath = VSLatest();
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Latest")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Latest")]
+        public static string VSLatest(this ICakeContext context)
+        {
+            var settings = new VSWhereLatestSettings();
+            return VSLatest(context, settings);
+        }
+
+        /// <summary>
+        /// Gets the latest Visual Studio product installation path.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>The Visual Studio installation path.</returns>
+        /// <example>
+        /// <code>
+        ///      var latestInstallationPath = VSLatest(new VSWhereLatestSettings { Requires = "'Microsoft.Component.MSBuild" });
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Latest")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Latest")]
+        public static string VSLatest(this ICakeContext context, VSWhereLatestSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            var latest = new VSWhereLatest(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools);
+            return latest.Latest(settings);
+        }
+
+        /// <summary>
+        /// Gets all Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSAll();
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("All")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.All")]
+        public static IEnumerable<string> VSAll(this ICakeContext context)
+        {
+            var settings = new VSWhereAllSettings();
+            return VSAll(context, settings);
+        }
+
+        /// <summary>
+        /// Gets all Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSAll(new VSWhereAllSettings { Requires = "'Microsoft.Component.MSBuild" });
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("All")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.All")]
+        public static IEnumerable<string> VSAll(this ICakeContext context, VSWhereAllSettings settings)
```

Comment:
Why did we go from VSWhereAll to VSAll?  Same for the other aliases.  I would have thought that the former would make sense.